### ISSUE TITLE
Detailed Moderation Purposed Logs

### ISSUE THREAD

**Issue by 989onan** [Yellow-Dog-Man/Resonite-Issues#1406]
### Is your feature request related to a problem? Please describe.

Currently Resonite has a problem where if you give someone permissions to do something in a session (ex: builder), they can make changes to a world without logging it. The current permissions system is great for me, and even allows some custom roles by modifying the component using non intended means. 
Usually when users are in a session that I'm hosting, it's fine if something breaks since I tell people to save often, and usually no lasting damage is done.

Unfortunately, this is rarely ever the case. People's stuff gets deleted sometimes, something """blows up""", or something gets broken. And sometimes the world auto saves and makes it irrecoverable. Not all of these cases means that someone was being malicious, but when code goes rogue, it can be impossible to tell what went wrong when the entire thing that caused it may have erased itself from existence.

### Describe the solution you'd like

What would be a good solution is to allow for logging actions that can be done by ProtoFlux and users. It can be a separate log that can be sent to moderation when making a ticket about an infraction or incident, or be used by hosts to find where something went wrong in a session. And maybe having multiple users in a session send their logs will allow moderation to rule out tampering
It would be nice for the log to monitor things like:
* Network connections or large packet spikes
* ProtoFlux nodes that can potentially cause damage (Parenting, moving, deleting, anchoring, knockback, setting transforms)
* actions done in inspectors by users in a session
* grabbing objects and moving them
* usage of tools and actions done with them to certain objects
* tooltip simulator usage
* FrooxEngine context changes in protoflux nodes, which might be logged using a TraceBack exception log type system (Aka, fire on change for host on a boolean toggle to make the host execute the code, and hopefully trace back the components and nodes that caused it)
* Sources of all of the log events and which client caused them

### Describe alternatives you've considered

Not giving users builder, or checking logs for who joined or left a session. Also using flux to try to relaliate against changes or mods to the client that prevent the function of certain nodes from working as intended (AKA NoKnockBack mod)

### Requesters

(Paste the text below and it will auto format for discord)
<@434468177062133772>
<@257663183966306304>

**Comment by 989onan** [comment]
After looking at this much later I can agree this is a kind of derivative.

It's an XY problem. Hard permissions have explicitly said that we will be able to tag world elements or items as unusable or unchangeable at the data model level. 

This means there is a high likely hood for the new system to prevent people from modifying stuff even with builder. you give someone permission with the ability to edit everything like builder, the data model can still block someone with unlimited power from modifying a slot or refID because it won't even accept the packet. (And that their role blacklists elements/slots tagged as such)

Im probably not fully correct and the new system may be different in some ways, but it would completely remove the need for this since we can specifically control what users can do rather than giving sweeping permissions


TL;DR I'll close this, since data model rework will make this irrelevant
